src/common/snippets/src/lowered/pass/assign_registers.cpp

```diff
@@ -46,12 +46,20 @@ bool AssignRegisters::run(LinearIR& linear_ir) {
     for (const auto& expr : expressions) {
         auto op = expr->get_node();
         if (const auto io_expr = std::dynamic_pointer_cast<IOExpression>(expr)) {
-            if (io_expr->get_type() == IOExpression::io_type::INPUT)
-                manually_assigned_gprs[expr->get_output_port_connector(0)] = io_expr->get_index();
-            else if (io_expr->get_type() == IOExpression::io_type::OUTPUT)
+            if (io_expr->get_type() == IOExpression::io_type::INPUT) {
```

Comment:
I just am not sure that we are ready to add support of these specific expressions. Seems like for them there should be common solution instead of a lot of `if`

Reply:
Added a comment with ticket number. Let's fix this while supporting full-scale Reshape.